lib/routemaster/middleware/payload_filter.rb

```diff
@@ -0,0 +1,10 @@
+module Routemaster
+  module Middleware
+    class PayloadFilter
+      # Filters duplicate events by url and type in a single payload.
+      def run(payload)
+        payload.group_by { |event| [event['url'], event['type']] }.map { |_, events| events.last }
```

Comment:
I don't think I am. `events.last` could just as easily be `events.first` and it wouldn't make a difference (apart from some very slight change in `t`)